src/main/java/com/puppycrawl/tools/checkstyle/site/XdocsTemplateSink.java

```diff
@@ -49,6 +58,42 @@ public class XdocsTemplateSink extends XdocSink {
     public XdocsTemplateSink(Writer writer, String encoding) {
         super(writer);
         this.encoding = encoding;
+        this.writer = new PrintWriter(writer);
+    }
+
+    /**
+     * Customization of org.apache.maven.doxia.sink.impl.XhtmlBaseSink.unifyEOL() method.
+     *
+     * @param text the text to scan.
+     * @return a String that contains only Linux System EOLs.
+     */
+    private static String unifyToLinuxEndOfLine(String text) {
+        return text
+                .replace("\r\n", LINUX_EOL)
+                .replace('\r', LINUX_EOL.charAt(0));
+    }
+
+    /**
+     * Overriden from org.apache.maven.doxia.sink.impl.Xhtml5BaseSink class.
+     * Ommited the usage of tableContentWriterStack, tableCaptionWriterStack and
+     * tableCaptionXMLWriterStack lists, since they are redundant in our current
+     * context and causing too many CI failures which includes PMD, Spotbugs and
+     * circleci - run inspections.
+     *
+     * @param text the text to write to the destination.
+     */
+    @Override
+    protected void write(String text) {
+        writer.write(unifyToLinuxEndOfLine(text));
+    }
```

Comment:
We can just use a compiled pattern instead of multiple replacements, this will match any line break, such as `\r`, `\r\n`, ` \u2028`, `\u2029` (any Unicode line break).

field should be:
```
Pattern LINE_BREAK_ESCAPE = Pattern.compile("\\R");
```

Then do:
```suggestion
    protected void write(String text) {
        final String newlinesOnlyText = LINE_BREAK_ESCAPE.matcher(text).replaceAll("\n");
        writer.write(unifyToLinuxEndOfLine(newlinesOnlyText));
    }
```